packages/slice-machine/src/modules/pushChangesSaga/index.ts

```diff
@@ -26,24 +26,59 @@ import {
 import { LoadingKeysEnum } from "../loading/types";
 import { ApiError } from "@src/models/ApiError";
 import { SyncError } from "@src/models/SyncError";
+import { ModelStatusInformation } from "@src/hooks/useModelStatus";
+import { ModelStatus } from "@lib/models/common/ModelStatus";
+import Tracker from "@src/tracking/client";
+
+const startTimer =
+  (startTime = Date.now()) =>
+  (endTime = Date.now()) =>
+    endTime - startTime;
 
 export const changesPushCreator = createAction("PUSH_CHANGES")<{
   unSyncedSlices: ReadonlyArray<ComponentUI>;
   unSyncedCustomTypes: ReadonlyArray<CustomTypeSM>;
+  modelStatuses: ModelStatusInformation["modelsStatuses"];
   onChangesPushed: (pushed: string | null) => void;
   handleError: (e: SyncError | null) => void;
 }>();
 
 export function* changesPushSaga({
   payload,
 }: ReturnType<typeof changesPushCreator>): Generator {
-  const { unSyncedSlices, unSyncedCustomTypes, onChangesPushed, handleError } =
-    payload;
+  const {
+    unSyncedSlices,
+    unSyncedCustomTypes,
+    modelStatuses,
+    onChangesPushed,
+    handleError,
+  } = payload;
   const totalNumberOfChanges: number =
     unSyncedSlices.length + unSyncedCustomTypes.length;
 
   let alreadySyncedChanges = 0;
   let stop: Parameters<typeof handleError>[0] = null;
+  let slicesCreated = 0;
```

Comment:
Do you think this could all go in a function outside that we just call at the beginning and return helpers to be called here ?

I feel this would remove a lot of complexity from the main saga.

Reply:
The variables?